src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -3731,6 +3731,21 @@ ReceiveResults(WorkerSession *session, bool storeRows)
 					{
 						executionStats->totalIntermediateResultSize += valueLength;
 					}
+
+					/*
+					 * We use this to count the amount of data that has been
+					 * received for EXPLAIN ANALYZE.
+					 * Only EXPLAIN ANALYZE TupleDestination has originalTask
+					 * defined. So that's why we check for it, otherwise we
+					 * don't have to keep track of this data.
+					 * The worker plan itself is also sent as a result in the
+					 * same task. We filter this out by only counting the data
+					 * from the first query.
+					 */
+					if (tupleDest->originalTask && queryIndex == 0)
+					{
+						tupleDest->originalTask->totalReceivedData += valueLength;
```

Comment:
Setting something in `struct Task` is tricky, since they are created once for prepared statements.

More specifically in this case, if you do the following:

```
create table t(a int, b int);
SELECT create_distributed_table('t', 'a');
prepare p1 as select * from t;
explain analyze execute p1;
explain analyze execute p1;
explain analyze execute p1;
```

You will get increasing data size in subsequent data sizes.

Reply:
Good catch on the EXPLAIN ANALYZE EXECUTE of a prepared statement without argumunts. I guess we need to reset the counter then somewhere. Either before filling it in, or probably better at cancellation and finish of the execution.

I agree on the encapsulation. What do you think of instead adding an optional `trackReceivedData(queryIndex, valueLength)` callback to the `tupleDest`, that would (currently) only be implemented by `ExplainAnalyzeDestination`. That way this if could be implemented there, which gives much better context.

Reply:
Not sure what the best approach is, one alternative approach is to add the `int32 tupleLibpqSize` argument to `putTuple`.

Also it might be nice to create a `struct ExplainAnalyzePrivate` (private to multi_explain.c) and put all `totalReceivedData`, `fetchedExplainAnalyzePlan`, and `fetechedExplainAnalyzePlacementIndex` there, and put an opaque pointer to it in `struct Task`.

Then it is clear that it is a bit more clear that this is the only member of `struct Task` that can safely be altered from execution to execution. Then create a `ResetExplainAnalyzeData(List *taskList)` and call it before calling `ExplainAnalyzeTaskList()` in adaptive_executor.c.